codec/legacy/codec.go

```diff
@@ -29,3 +30,9 @@ func PubKeyFromBytes(pubKeyBytes []byte) (pubKey cryptotypes.PubKey, err error)
 	err = Cdc.UnmarshalBinaryBare(pubKeyBytes, &pubKey)
 	return
 }
+
+// AminoPubKeyFromBytes unmarshals public key bytes and returns a PubKey
+func AminoPubKeyFromBytes(pubKeyBytes []byte) (pubKey multisig.LegacyAminoPubKey, err error) {
+	err = Cdc.UnmarshalBinaryBare(pubKeyBytes, &pubKey)
+	return
+}
```

Comment:
I don't think we want this function. Could you explain why adding it? 
My point is that we shouldn't create functions for specific types. It won't save amount of code:
```
// instead of calling AminoPubKeyFromBytes we can just write normally:
var pubkey multisig.LegacyAminoPubKey
if err = Cdc.UnmarshalBinaryBare(pubKeyBytes, &pubKey); err != nil
```

Reply:
The brutal, untested removal of this function caused the mess explained in #8633. Yes, your solution could work indeed. And yet, I believe that there is value in placing and documenting all amino-related legacy code in the `codec/legacy` package. So that when the time comes for us to remove all legacy stuff, we'll find it more easily. Wdyt?